xcdat/spatial_avg.py

```diff
@@ -324,83 +345,136 @@ def _get_weights(
             if domain_bounds is not None:
                 dom_bounds = domain_bounds.copy()
 
+                # validate domain bounds
+                self._validate_domain_bounds(dom_bounds)
+
                 if region_bounds is not None:
                     reg_bounds = np.array(region_bounds).astype(float)
                     if dim == "lon":
-                        # Forces longitude bounds linearity for elements where
-                        # the upper value is greater than the lower value (e.g.,
-                        # across prime meridian or dateline) to avoid issues
-                        # in operations/calculations that operate using the lon
-                        # bounds.
-                        dom_bounds, reg_bounds = self._force_lon_linearity(
```

Comment:
This use to try to make sure that the `dom_bounds` and `reg_bounds` lined up (but in specific cases did not work). This refactoring ensures that both `dom_bounds` and `reg_bounds` are in a 0 -> 360 coordinate system (the new relevant function is `_align_longitude_to_360_axis`).